From: YOKOTA Hiroshi <yokota.hgml@gmail.com>
Date: Fri, 9 Oct 2020 20:31:41 +0900
Subject: Use proper library file name

Forwarded: not-needed
---
 makefile | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/makefile b/makefile
index 1d4f2ea..94dc7b7 100644
--- a/makefile
+++ b/makefile
@@ -64,7 +64,9 @@ lib:	WHAT=RARDLL
 lib:	CXXFLAGS+=$(LIBFLAGS)
 lib:	$(OBJECTS) $(LIB_OBJ)
 	@rm -f libunrar.*
-	$(LINK) -Wl,-soname,libunrar.so.5 -shared -o libunrar.so $(LDFLAGS) $(OBJECTS) $(LIB_OBJ)
+	$(LINK) -Wl,-soname,libunrar.so.5 -shared -o libunrar.so.5.0 $(LDFLAGS) $(OBJECTS) $(LIB_OBJ)
+	ln -s libunrar.so.5.0 libunrar.so.5
+	ln -s libunrar.so.5.0 libunrar.so
 	$(AR) rcs libunrar.a $(OBJECTS) $(LIB_OBJ)
 
 install-unrar:
