From: YOKOTA Hiroshi <yokota.hgml@gmail.com>
Date: Sat, 10 Oct 2020 05:29:49 +0900
Subject: No need PIC option to static lib

Forwarded: not-needed
---
 makefile | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/makefile b/makefile
index 94dc7b7..1b69af4 100644
--- a/makefile
+++ b/makefile
@@ -63,10 +63,14 @@ sfx:	$(OBJECTS)
 lib:	WHAT=RARDLL
 lib:	CXXFLAGS+=$(LIBFLAGS)
 lib:	$(OBJECTS) $(LIB_OBJ)
-	@rm -f libunrar.*
+	@rm -f libunrar.so.5.0 libunrar.so.5 libunrar.so
 	$(LINK) -Wl,-soname,libunrar.so.5 -shared -o libunrar.so.5.0 $(LDFLAGS) $(OBJECTS) $(LIB_OBJ)
 	ln -s libunrar.so.5.0 libunrar.so.5
 	ln -s libunrar.so.5.0 libunrar.so
+
+static-lib: WHAT=RARDLL
+static-lib: $(OBJECTS) $(LIB_OBJ)
+	@rm -f libunrar.a
 	$(AR) rcs libunrar.a $(OBJECTS) $(LIB_OBJ)
 
 install-unrar:
