From: YOKOTA Hiroshi <yokota.hgml@gmail.com>
Date: Sun, 1 Nov 2020 03:24:45 +0900
Subject: Display build commands

Forwarded: not-needed
---
 makefile | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/makefile b/makefile
index e1c9717..84a8afb 100644
--- a/makefile
+++ b/makefile
@@ -40,37 +40,37 @@ install:	install-unrar
 uninstall:	uninstall-unrar
 
 clean:
-	@rm -f *.bak *~
-	@rm -f $(OBJECTS) $(UNRAR_OBJ) $(LIB_OBJ)
+	rm -f *.bak *~
+	rm -f $(OBJECTS) $(UNRAR_OBJ) $(LIB_OBJ)
 
 distclean: clean
-	@rm -f unrar libunrar.*
+	rm -f unrar libunrar.*
 
 # We removed 'clean' from dependencies, because it prevented parallel
 # 'make -Jn' builds.
 
 unrar:	$(OBJECTS) $(UNRAR_OBJ)
-	@rm -f unrar
+	rm -f unrar
 	$(LINK) -o unrar $(LDFLAGS) $(OBJECTS) $(UNRAR_OBJ) $(LIBS)	
 	$(STRIP) unrar
 
 sfx:	WHAT=SFX_MODULE
 sfx:	$(OBJECTS)
-	@rm -f default.sfx
+	rm -f default.sfx
 	$(LINK) -o default.sfx $(LDFLAGS) $(OBJECTS)
 	$(STRIP) default.sfx
 
 lib:	WHAT=RARDLL
 lib:	CXXFLAGS+=$(LIBFLAGS)
 lib:	$(OBJECTS) $(LIB_OBJ)
-	@rm -f libunrar.so.5.0 libunrar.so.5 libunrar.so
+	rm -f libunrar.so.5.0 libunrar.so.5 libunrar.so
 	$(LINK) -Wl,-soname,libunrar.so.5 -shared -o libunrar.so.5.0 $(LDFLAGS) $(OBJECTS) $(LIB_OBJ)
 	ln -s libunrar.so.5.0 libunrar.so.5
 	ln -s libunrar.so.5   libunrar.so
 
 static-lib: WHAT=RARDLL
 static-lib: $(OBJECTS) $(LIB_OBJ)
-	@rm -f libunrar.a
+	rm -f libunrar.a
 	$(AR) rcs libunrar.a $(OBJECTS) $(LIB_OBJ)
 
 install-unrar:
